experimental/tools/handshake-simulator/ExecModels.cpp

```diff
@@ -162,207 +261,541 @@ LogicalResult dynamatic::experimental::initialiseMap(
 namespace dynamatic {
 namespace experimental {
 
-/// Default CIRCT fork
-bool DefaultFork::tryExecute(llvm::DenseMap<Value, llvm::Any> &valueMap,
-                             llvm::DenseMap<unsigned, unsigned> &memoryMap,
-                             llvm::DenseMap<Value, double> &timeMap,
-                             std::vector<std::vector<llvm::Any>> &store,
-                             SmallVector<Value> &scheduleList, ModelMap &models,
-                             circt::Operation &opArg) {
+//--- Default CIRCT models ---------------------------------------------------//
 
+bool DefaultFork::tryExecute(ExecutableData &data, circt::Operation &opArg) {
   auto op = dyn_cast<circt::handshake::ForkOp>(opArg);
   auto executeFunc = [](std::vector<llvm::Any> &ins,
                         std::vector<llvm::Any> &outs, circt::Operation &op) {
     for (auto &out : outs)
       out = ins[0];
   };
-  return tryToExecute(op.getOperation(), valueMap, timeMap, scheduleList,
-                      models, executeFunc, 1);
+  return tryToExecute(op.getOperation(), data.valueMap, data.models,
+                      executeFunc);
 }
 
-/// Default CIRCT merge
-bool DefaultMerge::tryExecute(llvm::DenseMap<Value, llvm::Any> &valueMap,
-                              llvm::DenseMap<unsigned, unsigned> &memoryMap,
-                              llvm::DenseMap<Value, double> &timeMap,
-                              std::vector<std::vector<llvm::Any>> &store,
-                              SmallVector<Value> &scheduleList,
-                              ModelMap &models, circt::Operation &opArg) {
+bool DefaultMerge::tryExecute(ExecutableData &data, circt::Operation &opArg) {
   auto op = dyn_cast<circt::handshake::MergeOp>(opArg);
   bool found = false;
   for (Value in : op.getOperands()) {
-    if (valueMap.count(in) == 1) {
+    if (data.valueMap.count(in) == 1) {
       if (found)
         op.emitOpError("More than one valid input to Merge!");
-      auto t = valueMap[in];
+      auto t = data.valueMap[in];
 
-      valueMap[op.getResult()] = t;
-      timeMap[op.getResult()] = timeMap[in];
+      data.valueMap[op.getResult()] = t;
       // Consume the inputs.
-      valueMap.erase(in);
+      data.valueMap.erase(in);
       found = true;
     }
   }
   if (!found)
```

Comment:
I guess this is no longer an error now that we just try to execute all operations in a loop in all cycles.